modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/KeyValueProcessor.java

```diff
@@ -105,7 +111,9 @@ public void execute(IngestDocument document) {
                 }
                 return kv;
             })
-            .filter((p) -> includeKeys == null || includeKeys.contains(p[0]))
+            .filter((p) ->
+                (includeKeys == null || includeKeys.contains(p[0])) &&
+                    (excludeKeys == null || excludeKeys.contains(p[0]) == false))
```

Comment:
Those contains calls have a linear complexity due to the fact we are using lists, let's use hash sets instead? (and maybe even automata as a follow-up?)